src/DssGroupedEmergencySpell.sol

```diff
@@ -0,0 +1,129 @@
+    ///      if used incorrectly.
+    bytes32[] private ilkList;
+
+    /// @param _ilks The list of ilks for which the spell should be applicable
+    /// @dev The list size is be at least 2 and less than or equal to 3.
+    ///      The grouped spell is meant to be used for ilks that are a variation of tha same collateral gem
+    ///      (i.e.: ETH-A, ETH-B, ETH-C)
+    ///      There has never been a case where MCD onboarded 4 or more ilks for the same collateral gem.
+    ///      For cases where there is only one ilk for the same collateral gem, use the single-ilk version.
+    constructor(bytes32[] memory _ilks) {
+        // This is a workaround to Solidity's lack of support for immutable arrays, as described in
+        // https://github.com/ethereum/solidity/issues/12587
+        uint256 len = _ilks.length;
+        require(len >= MIN_ILKS, "DssGroupedEmergencySpell/too-few-ilks");
+
+        ilkList = _ilks;
+    }
+
+    /// @notice Returns the list of ilks to which the spell is applicable.
+    function ilks() external view returns (bytes32[] memory) {
+        return ilkList;
+    }
+
+    /// @notice Returns the spell description.
+    function description() external view returns (string memory) {
+        // Join the list of ilks into a comma-separated string
+        string memory buf = _bytes32ToString(ilkList[0]);
+        // Start from one because the first item was already added.
+        for (uint256 i = 1; i < ilkList.length; i++) {
+            buf = string.concat(buf, ", ", _bytes32ToString(ilkList[i]));
+        }
+
+        return string.concat(_descriptionPrefix(), " ", buf);
+    }
+
+    /// @notice Converts a bytes32 value into a string.
+    function _bytes32ToString(bytes32 src) internal pure returns (string memory res) {
+        uint256 len = 0;
+        while (src[len] != 0 && len < 32) {
+            len++;
+        }
+        assembly {
+            res := mload(0x40)
+            // new "memory end" including padding (the string isn't larger than 32 bytes)
+            mstore(0x40, add(res, 0x40))
+            // store len in memory
+            mstore(res, len)
+            // write actual data
+            mstore(add(res, 0x20), src)
+        }
+    }
+
+    /// @dev Returns the description prefix to compose the final description.
+    function _descriptionPrefix() internal view virtual returns (string memory);
+
+    /// @inheritdoc DssEmergencySpell
+    function _emergencyActions() internal override {
+        for (uint256 i = 0; i < ilkList.length; i++) {
+            _emergencyActions(ilkList[i]);
+        }
+    }
+
+    /// @notice Executes the emergency actions for all ilks in the batch.
+    /// @dev This is an escape hatch to prevent the spell from being blocked in case it would hit the block gas limit.
+    ///      In case `end` is greater than the ilk registry length, the iteration will be automatically capped.
```

Comment:
```suggestion
    ///      In case `end` is greater than the ilk list length, the iteration will be automatically capped.
```
Differentiate between the ilk list and the external registry